public/modules/custom/grants_metadata/src/Controller/ApplicationStatusCheckController.php

```diff
@@ -18,76 +17,51 @@
 class ApplicationStatusCheckController extends ControllerBase {
 
   /**
-   * The helfi_atv service.
+   * The config.
    *
-   * @var \Drupal\helfi_atv\AtvService
+   * @var \Drupal\Core\Config\ImmutableConfig
    */
-  protected AtvService $helfiAtv;
-
-  /**
-   * Helsinkiprofiili service.
-   *
-   * @var \Drupal\helfi_helsinki_profiili\HelsinkiProfiiliUserData
-   */
-  protected HelsinkiProfiiliUserData $helsinkiProfiiliUserData;
-
-  /**
-   * The language manager.
-   *
-   * @var \Drupal\Core\Language\LanguageManagerInterface
-   */
-  protected $languageManager;
-
-  /**
-   * The config factory.
-   *
-   * @var \Drupal\Core\Config\ConfigFactoryInterface
-   */
-  protected $configFactory;
+  protected ImmutableConfig $config;
 
   /**
    * The controller constructor.
    *
-   * @param \Drupal\helfi_atv\AtvService $helfi_atv
+   * @param \Drupal\helfi_atv\AtvService $helfiAtv
    *   The helfi_atv service.
    * @param \Drupal\helfi_helsinki_profiili\HelsinkiProfiiliUserData $helsinkiProfiiliUserData
    *   The helfi_atv service.
-   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
-   *   Config factory.
-   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
-   *   Language manager.
    */
   public function __construct(
-    AtvService $helfi_atv,
-    HelsinkiProfiiliUserData $helsinkiProfiiliUserData,
-    ConfigFactoryInterface $configFactory,
-    LanguageManagerInterface $language_manager,
+    protected AtvService $helfiAtv,
+    protected HelsinkiProfiiliUserData $helsinkiProfiiliUserData,
   ) {
-    $this->helfiAtv = $helfi_atv;
-    $this->helsinkiProfiiliUserData = $helsinkiProfiiliUserData;
-    $this->configFactory = $configFactory;
-    $this->languageManager = $language_manager;
+    $this->config = $this->configFactory->get('grants_handler.settings');
   }
 
   /**
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container) {
-    return new static(
+  public static function create(ContainerInterface $container):static {
+    return new self(
```

Comment:
The return type hit (`static`) does not match with the actual return type (`self`)